onnx/shape_inference.py

```diff
@@ -82,4 +82,42 @@ def infer_shapes_path(
         )
 
 
+def infer_node_outputs(
+    schema: onnx.defs.OpSchema,
+    node: onnx.NodeProto,
+    input_types: Dict[str, onnx.TypeProto],
```

Comment:
Do we explicitly require a dictionary for these params? If not prefer a more generic type like Mapping. 

——-

Use more generic types for input parameters and specific types for return values. For example, a function may take a `Sequence` and return a `List`.

Diagram for reference: https://gist.github.com/justinchuby/4021cebe9e093f636759a88de325c85f